src/components/dialog/dialog.component.js

```diff
@@ -29,42 +30,42 @@ class Dialog extends Modal {
     super.componentDidMount();
     if (this.props.open) {
       this.centerDialog();
-      if (this.props.autoFocus) {
-        this.focusDialog();
-      }
     }
   }
 
   onDialogBlur(ev) { } // eslint-disable-line no-unused-vars
 
-  onCloseIconBlur(ev) {
-    ev.preventDefault();
-    this.focusDialog();
-  }
-
-  get onOpening() {
+  onOpening() {
     this.document.documentElement.style.overflow = 'hidden';
     this.centerDialog(true);
     ElementResize.addListener(this._innerContent, this.applyFixedBottom);
     this.window.addEventListener('resize', this.centerDialog);
-
-    if (this.props.autoFocus) {
-      return this.focusDialog();
-    }
-
-    return null;
+    this.focusTrap = new FocusTrap(this._dialog);
+    this.focusTrap.setFocusTrap();
+    this.focusTrap.setFocusToFirstFocusableElement(this.props.firstFocus);
```

Comment:
I see the user now has to do `<Dialog focusFirst>` I know this was added for the case when a user wants to focus the first element. However, I think we should simplify this by keeping the existing `<Input autoFocus>` props.

At the moment if someone had 

```
<Dialog focusFirst>
<Input>
<Input autoFocus>
</Dialog>
```
 it would cause problems, I think the interface is simpler if we remove the focus functionality from `focusTrap`. What do you think?

Reply:
I have removed first focus